src/NuGet.Core/NuGet.ProjectManagement/NuGetProject.cs

```diff
@@ -35,6 +37,14 @@ protected NuGetProject(Dictionary<string, object> metadata)
             get { return InternalMetadata; }
         }
 
+        public string Guid
```

Comment:
> Idea here is to map a start batch event with corresponding end batch event

That makes perfect sense, but the NuGetProject objects are created once when the solution is opened. Every batch event for a project will have the same guid, why define it on the project at all?